__tests__/test-utils.js

```diff
@@ -103,16 +117,62 @@ class TestUtils {
     return request;
   }
 
+  static async walletStatus({ walletId = WALLET_ID, firstAddress = null }) {
+    const params = { 'x-wallet-id': walletId };
+    if (firstAddress) {
+      params['x-first-address'] = firstAddress;
+    }
+    const response = await request.get('/wallet/status').set(params);
+    TestUtils.logger.debug('[TestUtil:walletStatus] wallet status', { walletId, body: response.body });
+    return response;
+  }
+
+  static async waitReady({
+    walletId = WALLET_ID,
+    exitIfClosed = false,
+    retries = 3,
+    firstAddress = null,
+  } = {}) {
+    for (let i = 0; i < retries; i++) {
+      const res = await TestUtils.walletStatus({ walletId, firstAddress });
+      if (res.body?.success !== false) {
+        return true;
+      }
+      if (res.body?.message === 'Invalid wallet id parameter.') {
+        // The wallet does not exist
+        return false;
+      }
+      if (res.body?.statusCode === HathorWallet.ERROR) {
+        throw new Error(res.body?.message);
+      }
+      if (exitIfClosed && res.body?.statusCode === HathorWallet.CLOSED) {
+        return false;
+      }
+      await new Promise(resolve => {
+        setTimeout(resolve, 500);
+      });
```

Comment:
We have a `delay` function on [core.util.js](https://github.com/HathorNetwork/hathor-wallet-headless/blob/master/__tests__/integration/utils/core.util.js#L8-L17) that I think would be better than calling `setTimeout` here, for simplicity.

Maybe we should open a new issue later to move the `core.util` file to the main tests folder, instead of a "integration test" specific one. It's a small change, but a good one to keep things tidy.

Reply:
I saw the delay method but I wanted to keep the same execution as the `startWallet`. Maybe on another PR we can make an actual utility core to be used on both integration and unit tests.